src/helpers.js

```diff
@@ -0,0 +1,49 @@
+import * as firebase from "firebase/app";
+import "firebase/auth";
+import "firebase/firestore";
+import { firebaseConfig } from "./constants";
+
+firebase.initializeApp(firebaseConfig);
+
+const db = firebase.firestore();
+const collectionRef = db.collection("carts");
+const getQuery = db.collectionGroup("carts").orderBy("timestamp", "asc");
+console.log("getQuery", getQuery);
+
+export const getCarts = callback => {
+  getQuery
+    .get()
+    .then(response => {
+      const respJson = response.docs.map(
+        doc => doc && doc.exists && doc.data()
+      );
+      callback(respJson);
+    })
+    .catch(error => {
+      console.error("Error retrieving carts from database: ", error);
+    });
+};
+
+getCarts(resp => {
+  console.log(resp);
+});
+
+export const onCartsUpdate = callback => {
```

Comment:
This looks more like a listener setup function than on `onEvent` method

Reply:
Changed name